blueocean-dashboard/src/main/js/util/UrlUtils.js

```diff
@@ -173,11 +173,12 @@ export function getRestUrl({ organization, pipeline, branch, runId }) {
     const jenkinsUrl = require('../config').getJenkinsRootURL();
     let url = `${jenkinsUrl}/blue/rest/organizations/${encodeURIComponent(organizationName)}`;
     if (pipelineName) {
-        // pipelineName might include a folder path, don't escape it
+        // pipelineName might include a folder path, don't encode it
         url += `/pipelines/${pipelineName}`;
     }
     if (branch) {
-        url += `/branches/${encodeURIComponent(branch)}`;
+        // JENKINS-37712 branch needs to be double-encoded for some reason
+        url += `/branches/${encodeURIComponent(encodeURIComponent(branch))}`;
```

Comment:
This is going to work as long as branch name is not already encoded. For example, for multi-branch project /pipelines/:id/ API will give branch names as an array:

    "branchNames" : [ "master", "feature2", "feature%2Fux-1" ]

Above 'feature/ux-1' appears encoded. If UI code uses this name and calls getResstUrl() its going to break. Maybe these branch names are coming from SSE events and are not encoded? Maybe then documenting this method that branch must not be encoded?